osbs/api.py

```diff
@@ -713,6 +713,57 @@ def get_build_logs(self, build_id, follow=False, build_json=None, wait_if_missin
 
         return logs
 
+    @staticmethod
+    def _parse_build_log_entry(entry):
+        items = entry.split()
+        if len(items) < 4:
+            logger.warning("This is not a valid build log entry: %s", entry)
+            return (None, entry)
+
+        platform = items[2]
+        if not platform.startswith("platform:"):
+            logger.warning("Line logged without using the appropriate LoggerAdapter")
+            return (None, entry)
+
+        platform = platform.split(":", 1)[1]
+        if platform == "-":
+            return (None, entry)  # proper orchestrator build log entry
+
+        # Anything else should be a worker build log entry, so we strip off
+        # the leading 8 wrapping orchestrator log fields:
+        # <date> <time> <platform> - <name> - <level> -
+        plen = sum(len(items[i]) + 1  # include trailing space
+                   for i in range(8))
+        line = entry[plen:]
+        # if the 3rd field is "platform:-", we strip it out
+        items = line.split()
+        if items[2] == "platform:-":
+            plen = sum(len(items[i]) + 1  # include trailing space
+                       for i in range(3))
+            line = "%s %s %s" % (items[0], items[1], line[plen:])
+        return (platform, line)
```

Comment:
The docs say this we'll yield an object with attributes `platform` and `line`. One way of doing that is to declare a namedtuple (at module scope):
```
from collections import namedtuple
LogEntry = namedtuple('LogEntry', ['platform', 'line'])
```

Then you can replace this return value with an instance of that new class:
```
return LogEntry(platform, line)
```